### PR TITLE
feat(challenge-email): Hide email auth code entry behind link

### PR DESCRIPTION
**Pull request by zposten-okta**
## Description:

This PR modifies the UI of the screen the user lands on after they have chosen to authenticate with a magic link.  It changes the text that is displayed to include the term "magic link" and to also include the user's email address when possible.

In addition to changing the language, the text field and "verify" button are hidden by default.  These are used to enter the code emailed to the user if their magic link didn't work for some reason.  Most of the time though we assume that the magic link will work, so hiding them by default simplifies the experience for most users.

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added (**updated**) unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

#### BEFORE

image

#### AFTER

default, hidden button

default, shown button

**Errors and warnings are place below the link, but above the button and text field**

error placement, hidden button

error placement, shown button

**When the user's email address is not available, "your email" is displayed.**

your email, hidden button

your email, shown button

### Reviewers:

@sherizada-okta 

### Issue:

- [OKTA-402914](https://oktainc.atlassian.net/browse/OKTA-402914)

### CODE REVIEW

**Comment by pradeepdewda-okta** [comment]
@zposten nice. Add PR description and add a video of the change.

**Comment by pradeepdewda-okta** [comment]
@zposten-okta this is going to break selenium tests in okta-core.. Before merge run downstream from this topic branch on bacon to see impact on okta-core

**Comment by pradeepdewda-okta** [comment]
@zposten @vincentvoong-okta  I feel this statement needs an improvement when we display the link instead of enter code field
```
Click the link in the email or enter code the code below to continue
```

something like below? because user will click the link instead of actually entering the code
```
Click the link in the email or click the link below to enter code instead
```

**Comment by zposten-okta** [comment]
Hey @pradeepdewda-okta, thank you so much for reviewing this PR!  I will definitely run the selenium tests in Okta core.  Still trying to get that to work in bacon currently.

Relating to your suggested text change, I totally see why you would say that.  They need to perform an action before they can actually enter the code.  I'm definitely willing to make whatever change is deemed necessary, but my viewpoint on this is that the first paragraph directs them "below" to the link, whose text I think is pretty clear saying "Enter a code from the email instead".  Just thinking out loud here, saying "click the link" twice I think is a little confusing because as a user I'm just trying to read as little text as possible to get the job done.

File: src/v2/controllers/FormController.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ export default Controller.extend({
       currentViewState.name,
       this.options.appState.get('authenticatorKey'),
     );
+
```
Comment (pradeepdewda-okta) [inline src/v2/controllers/FormController.js:49]: nit

File: src/v2/view-builder/views/email/ChallengeAuthenticatorEmailView.js [hunk 2]
```diff
@@ -11,35 +11,94 @@ const CheckYourEmailTitle = View.extend({
...
+        selector: '.o-form-error-container',
+        options: { email },
+      });
+    },
+
+    postRender() {
+      this.showCodeEntryField(false);
+    },
+
+    events: {
```
Comment (pradeepdewda-okta) [inline src/v2/view-builder/views/email/ChallengeAuthenticatorEmailView.js:77]: i would raise `events` above `initialize`

**Comment by pradeepdewda-okta** [comment]
> Hey @pradeepdewda-okta, thank you so much for reviewing this PR! I will definitely run the selenium tests in Okta core. Still trying to get that to work in bacon currently.
> 
> Relating to your suggested text change, I totally see why you would say that. They need to perform an action before they can actually enter the code. I'm definitely willing to make whatever change is deemed necessary, but my viewpoint on this is that the first paragraph directs them "below" to the link, whose text I think is pretty clear saying "Enter a code from the email instead". Just thinking out loud here, saying "click the link" twice I think is a little confusing because as a user I'm just trying to read as little text as possible to get the job done.

Now when i think like that.. It make sense. thanks.

**Review by pradeepdewda-okta — APPROVED** [review]

File: src/v2/view-builder/views/email/ChallengeAuthenticatorEmailView.js (hunk 2 above)
Comment (zposten-okta) [reply src/v2/view-builder/views/email/ChallengeAuthenticatorEmailView.js:77]: Are they supposed to be in alphabetical order?

Comment (pradeepdewda-okta) [reply src/v2/view-builder/views/email/ChallengeAuthenticatorEmailView.js:77]: No.. just all functions together and ideally `initialize` defined first.

**Pull request closed without merging**